promql/engine.go

```diff
@@ -1384,13 +1396,20 @@ func (ev *evaluator) eval(expr parser.Expr) (parser.Value, annotations.Annotatio
 			if ok {
 				return ev.rangeEvalTimestampFunctionOverVectorSelector(vs, call, e)
 			}
+		case "histogram_quantile", "quantile_over_time":
+			unwrapParenExpr(&e.Args[0])
+			arg := unwrapStepInvariantExpr(e.Args[0])
+			unwrapParenExpr(&arg)
+			q, _ := strconv.ParseFloat(arg.String(), 64)
```

Comment:
Same argument here as above: The arg could be a complex expression (or even change depending on the evaluation time).